src/concepts/kafka-proxies/http-proxy.md

```diff
@@ -0,0 +1,67 @@
+---
+description: Zilla lets you configure application-centric REST API and SSE stream endpoints that unlock Kafka event-driven architectures.
+prev: false
+next: /tutorials/rest/rest-intro.md
+---
+
+# HTTP Kafka Proxy
+
+Zilla lets you configure application-centric REST APIs and SSE streams that unlock Kafka event-driven architectures. A developer has the freedom to define their own HTTP mapping to Kafka, with control over the topics, message key, message headers, and payload. Any HTTP client can interact with Kafka without navigating Kafka-specific paradigms.
+
+## Configure Endpoints
+
+Zilla can map REST APIs to Kafka using the [http-kafka](../../reference/config/bindings/binding-http-kafka.md) binding in `zilla.yaml`. Zilla routes REST urls using [wildcard pattern matching](../../concepts/config-intro.md#pattern-matching) and [dynamic path params](../../concepts/config-intro.md#dynamic-path-parameters). Dynamic path matching and custom message routing from endpoints to Kafka topics help prevent API lock-in. 
+
+### HTTP request methods
+
+Zilla separates the HTTP request methods into two groups called capabilities: produce and fetch. The [produce](../../concepts/config-intro.md#the-fetch-capability) capability handles method types `POST`, `PUT`, `DELETE`, and `PATCH` that produces messages onto Kafka topics. The [fetch](../../reference/config/bindings/binding-http-kafka.md#with-capability-fetch) capability handles the `GET` method that fetches messages from Kafka topics. One exception is for a route managing async correlation. The `produce` route will have two when clauses: a `PUT` clause for submission and a `GET` clause matching the `async.location` path returned to the caller.
+
+## Correlated Request-Response
+
+Zilla manages the HTTP lifecycle with the request and response payload on the event stream over a pair of Kafka topics. Each request message is correlated to the corresponding response message with a `zilla:correlation-id` header, providing an identifier for both Zilla and Kafka workflows to act on.
+
+### sync
+
+A synchronous interaction from the client will open a connection, producing a request message. The connection will remain open, waiting for the correlated response message to be fetched and returned to the caller.
+
+### async
+
+An asynchronous interaction includes a `prefer: respond-async` header. After producing a request message, the connection will immediately return with `202 Accepted` plus the location path to retrieve a correlated response. The client then sends a request to the returned location path with the `prefer: wait=N` header to retrieve the correlated response as soon as it becomes available, removing the need for client polling.
```

Comment:
Since N is a placeholder it's worth calling out that Zilla will wait for up to N seconds.